### PR TITLE
Added removal of producer/owner/app labels and forcing of token labels

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/script/functions/FIND.java [hunk 1]
```diff
@@ -181,13 +181,20 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
...
         for (int i = 0; i < selectors.size(); i++) {
           String csel = (String) selectors.get(i).getLeft();
           Map<String,String> lsel = (Map<String,String>) selectors.get(i).getRight();
-          
+          lsel.remove(Constants.PRODUCER_LABEL);
+          lsel.remove(Constants.OWNER_LABEL);
+          lsel.remove(Constants.APPLICATION_LABEL);
+          lsel.putAll(Tokens.labelSelectorsFromReadToken(rtoken));
+
           drequest = new DirectoryRequest();
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/FIND.java]: Should be outside `for` loop.

File: warp10/src/main/java/io/warp10/script/functions/FIND.java [hunk 2]
```diff
@@ -181,13 +181,20 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     if (mapparams) {
       top = stack.pop();
       Map<String,Object> params = paramsFromMap((Map) top);
+
+      token = (String) params.get(FETCH.PARAM_TOKEN);
+      ReadToken rtoken = Tokens.extractReadToken(token);
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/FIND.java]: No need for that, see next comment.

File: warp10/src/main/java/io/warp10/script/functions/FIND.java [hunk 3]
```diff
@@ -181,13 +181,20 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
...
         List<Pair<Object, Object>> selectors = (List<Pair<Object, Object>>) params.get(FETCH.PARAM_SELECTOR_PAIRS);
         for (int i = 0; i < selectors.size(); i++) {
           String csel = (String) selectors.get(i).getLeft();
           Map<String,String> lsel = (Map<String,String>) selectors.get(i).getRight();
-          
+          lsel.remove(Constants.PRODUCER_LABEL);
+          lsel.remove(Constants.OWNER_LABEL);
+          lsel.remove(Constants.APPLICATION_LABEL);
+          lsel.putAll(Tokens.labelSelectorsFromReadToken(rtoken));
+
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/FIND.java]: No need for that, already done lines 327 to 330.

**Comment by hbs** [comment]
PTAL

**Pull request merged**